pkg/statsd/statsd.go

```diff
@@ -138,8 +139,12 @@ func (s *Server) RunWithCustomSocket(ctx context.Context, sf SocketFactory) erro
 	}
 
 	bufferSize := 1000 // Estimating this is hard, and tends to cause loss under adverse conditions
-	statser := stats.NewInternalStatser(bufferSize, s.InternalTags, namespace, hostname, metrics, events)
-	// TODO: Make internal metric dispatch configurable
+	var statser stats.Statser
+	if s.DisableInternalMetrics {
+		statser = stats.NewNullStatser()
+	} else {
+		statser = stats.NewInternalStatser(bufferSize, s.InternalTags, namespace, hostname, metrics, events)
+	}
 	// statser := stats.NewLoggingStatser(s.InternalTags, log.NewEntry(log.New()))
 	stage = stgr.NextStage()
```

Comment:
If you move these two lines into the place where you are constructing internal statser then you'd not need to add `Run()` to the interface and null statser implementation :) Much simpler. You'd need to use an intermediate variable to hold the internal statser to avoid loosing it's type immediately so that you can pass it to the stage.start.